blocks/blockstore/blockstore.go

```diff
@@ -71,6 +72,12 @@ type blockstore struct {
 	lk      sync.RWMutex
 	gcreq   int32
 	gcreqlk sync.Mutex
+
+	rehash bool
+}
+
+func (bs *blockstore) RuntimeHashing(enabled bool) {
```

Comment:
- this is not a clear name, because there _is_ runtime hashing regardless
- use: `HashOnRead`